Added a prompt before closing SteamVR

sync1211 opened the pull request:
Added a prompt asking whether the setup should close SteamVR when it detects that SteamVR is running.

This prevents the user from being thrown out of the game without warning and gives them a chance to shut down SteamVR gracefully.

sync1211 committed:
+ Ask if SteamVR should be closed during setup

sync1211 committed:
+ Only cancel setup if SteamVR is still running after the "Close SteamVR?" prompt
+ Updated the title of the prompt to close SteamVR

HimbeersaftLP commented on k2vr-installer-gui/Tools/Utils.cs:
```diff
@@ -20,40 +20,57 @@ public static bool IsDirectoryEmpty(string path)
             Logger.Log("Checking if SteamVR is open...", false);
-            foreach (Process process in Process.GetProcesses())
+
+            if (Process.GetProcesses().FirstOrDefault((Process proc) => proc.ProcessName == "vrserver" || proc.ProcessName == "vrserver") != null)
```
vrserver or vrserver?

sync1211 replied:
I'm sorry, there must have been an issue

HimbeersaftLP commented on k2vr-installer-gui/Tools/Utils.cs:
```diff
@@ -20,40 +20,57 @@ public static bool IsDirectoryEmpty(string path)
-                    Thread.Sleep(5000);
-                    if (!process.HasExited)
+                    //Only exit if SteamVR is still running
+                    if (Process.GetProcesses().FirstOrDefault((Process proc) => proc.ProcessName == "vrserver" || proc.ProcessName == "vrserver") != null)
```
Same here. Also UX of this is a problem. If I close SteamVR by hand, and then click "No" (thought process: I already closed it myself, I don't need this tool to do it for me), it will still close the installer.

HimbeersaftLP commented on k2vr-installer-gui/Tools/Utils.cs:
```diff
@@ -20,40 +20,57 @@ public static bool IsDirectoryEmpty(string path)
+            //Close VrMonitor
+            foreach (Process process in Process.GetProcesses().Where((proc) => proc.ProcessName == "vrmonitor"))
+            {
+                Logger.Log("Closing vrmonitor (PID: " + process.Id + ")...", false);
```
Is the PID really that relevant? (logs get displayed on the install screen, does the user care about the PID?)

sync1211 replied:
I though that it would be good to know the PID in case something goes wrong.

HimbeersaftLP replied:
But after it has been forcefully killed, is the PID going to help?

sync1211 replied:
Not really.

sync1211 committed:
* Fixed redundant check for vrserver
* Fixed missing check for vrmonitor
+ Simplified the checks for SteamVR processes
- Removed PIDs from logging messages

HimbeersaftLP commented on k2vr-installer-gui/Tools/Utils.cs:
```diff
@@ -20,40 +20,58 @@ public static bool IsDirectoryEmpty(string path)
+                MessageBoxImage.Warning
+                ) != MessageBoxResult.Yes)
+            {
+                //Only exit setup if SteamVR is still running after choosing "No"
```
Can you add a MessageBox that says something like "The installer will now close" before that if it does close, because otherwise it looks like a crash.

HimbeersaftLP commented on k2vr-installer-gui/Tools/Utils.cs:
```diff
@@ -20,40 +20,58 @@ public static bool IsDirectoryEmpty(string path)
+            //Check if SteamVR is running
+            if (Process.GetProcesses().FirstOrDefault((Process proc) => proc.ProcessName == "vrserver" || proc.ProcessName == "vrmonitor") == null)
+            {
+                return true;
```
Should log "Done!" before this return (and the other return on the second check, if it is true) for consistency

sync1211 committed:
+ Added a message box for when the user has chosen not to close SteamVR
* Fixed missing logging calls in EnsureSteamVrClosed

HimbeersaftLP reviewed (approved):
Perfect
